### ISSUE TITLE
Crash in datatransport com.google.android.datatransport.runtime.logging.Logging 

### ISSUE THREAD

**Issue by gevorg-kopalyan** [firebase/firebase-android-sdk#3716]
### [READ] Step 1: Are you in the right place?

yes

### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Android Studio Bumblebee | 2021.1.1 Patch 3
  * Firebase Component:  Analytics, Auth, Performance, Dynamic links, Crashlytics, Remote Config, Messaging 
  * Component version: Bom 30.0.0

#### Steps to reproduce:

Getting crashes in google play console but not in Firebase Crashlytics 

```
java.lang.IllegalArgumentException: Log tag "TransportRuntime.Executor" exceeds limit of 23 characters

  at android.util.Log.isLoggable (Native Method)
  at com.google.android.datatransport.runtime.logging.Logging.e (Logging.java:63)
  at com.google.android.datatransport.runtime.SafeLoggingExecutor$SafeLoggingRunnable.run (SafeLoggingExecutor.java:49)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1133)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:607)
  at java.lang.Thread.run (Thread.java:776)
```

In com.google.android.datatransport:transport-runtime Library which firebase datatrasport library is using has a new  Log.isLoggable() check which doesn't check for a tag limit of 23 characters and can produce crash.
Here is the [pull request](https://github.com/firebase/firebase-android-sdk/pull/3648) that added new Log.isLoggable check

**Comment by darrentaft** [comment]
Glad it's not just me! Had to immediately roll back to 29.3.1. Thanks for raising.

**Comment by argzdev** [comment]
Thanks for reporting, @gevorg-kopalyan, @darrentaft. I'll notify our engineers and see what we can do here.

**Comment by vlad-roid** [comment]
This is **bad**, it's hitting affected users repeatedly. So far only 11 people in our app are affected, but each person has been hit by the crash over 50 times! I assume this happens on app trying to auto-start, I can't imagine someone having the patience to try and open an app 50 times.

![image](https://user-images.githubusercontent.com/26101312/168507296-a4aedacc-61a3-4b72-9fa4-729467b6f469.png)

**Comment by AliA74** [comment]
I am facing the same issue. I know this issue has been fixed in the source code but how can I as a Unity game developer using Firebase-SDK fix this issue? Has there been any new release that I can use? Either for Android or Unity.

**Comment by argzdev** [comment]
Hi @AliA74, we expect this fix to be in a release that is targeted for next week. You can keep an eye on our [release notes](https://firebase.google.com/support/release-notes/android) for bug fixes and features. As for Firebase SDK for the Unity platform, afaik it should be available once it's been released for Android.

**Comment by aman-reso** [comment]
How you guy's resolve this Crash

**Comment by AliA74** [comment]
@aman-reso 

I used ProGuard to remove isLoggable calls. Add something similar to your proguard file:

`-assumenosideeffects class android.util.Log {
  static *** isLoggable(...);
}`

**Comment by carlonzo** [comment]
the fix for this has been released as part of BOM 30.0.2 https://firebase.google.com/support/release-notes/android#bom_v30-0-2
